### PR TITLE
Upgrade Python 3.11

### PR DESCRIPTION
**Pull request by dmichaels-harvard**

### CODE REVIEW

File: .github/workflows/main-publish.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-20.04
```
Comment (netsettler) [inline .github/workflows/main-publish.yml]: perhaps use 22.04?

File: Makefile [hunk 2]
```diff
@@ -19,6 +19,16 @@ help:
 clean:
 	rm -rf *.egg-info
 
+publish:
```
Comment (netsettler) [inline Makefile:23]: I can't comment on regions you didn't change, but a bit higher in the file than here, I would not mind seeing build do `make configure`.

**Review by netsettler — APPROVED** [review]
My comments are not blocking.

**Review by willronchetti — APPROVED** [review]

File: .github/workflows/main-publish.yml (hunk 1 above)
Comment (dmichaels-harvard) [reply .github/workflows/main-publish.yml]: Done

File: Makefile (hunk 2 above)
Comment (dmichaels-harvard) [reply Makefile:23]: Done

**Pull request merged**